owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/shares/CreateRemoteShareOperation.kt

```diff
@@ -81,90 +93,117 @@ class CreateRemoteShareOperation(
 
     var retrieveShareDetails = false // To retrieve more info about the just created share
 
-    override fun run(client: OwnCloudClient): RemoteOperationResult<ShareParserResult> {
-        var result: RemoteOperationResult<ShareParserResult>
+    private fun buildRequestUri(baseUri: Uri) =
+        baseUri.buildUpon()
+            .appendEncodedPath(OCS_ROUTE)
+            .appendQueryParameter(PARAM_FORMAT, VALUE_FORMAT)
+            .build()
+
+    private fun parseResponse(response: String): ShareResponse? {
+        val moshi = Moshi.Builder().build()
+        val commonOcsType: Type = Types.newParameterizedType(CommonOcsResponse::class.java, ShareItem::class.java)
+        val adapter: JsonAdapter<CommonOcsResponse<ShareItem>> = moshi.adapter(commonOcsType)
+        val remoteShare = adapter.fromJson(response)?.ocs?.data?.toRemoteShare()
+        return ShareResponse(remoteShare?.let { listOf(it) } ?: listOf())
+    }
 
-        try {
-            val formBodyBuilder = FormBody.Builder()
-                .add(PARAM_PATH, remoteFilePath)
-                .add(PARAM_SHARE_TYPE, shareType.value.toString())
-                .add(PARAM_SHARE_WITH, shareWith)
+    private fun onResultUnsuccessful(
+        method: PostMethod,
+        response: String?,
+        status: Int
+    ): RemoteOperationResult<ShareResponse> {
+        Timber.e("Failed response while while creating new remote share operation ")
+        if (response != null) {
+            Timber.e("*** status code: $status; response message: $response")
+        } else {
+            Timber.e("*** status code: $status")
+        }
+        return RemoteOperationResult(method)
+    }
 
-            if (name.isNotEmpty()) {
-                formBodyBuilder.add(PARAM_NAME, name)
-            }
+    private fun onRequestSuccessful(response: String?): RemoteOperationResult<ShareResponse> {
+        val result = RemoteOperationResult<ShareResponse>(RemoteOperationResult.ResultCode.OK)
+        Timber.d("Successful response: $response")
+        result.data = parseResponse(response!!)
+        Timber.d("*** Creating new remote share operation completed ")
+        return result
+    }
 
-            if (expirationDateInMillis > INIT_EXPIRATION_DATE_IN_MILLIS) {
-                val dateFormat = SimpleDateFormat(FORMAT_EXPIRATION_DATE, Locale.getDefault())
-                val expirationDate = Calendar.getInstance()
-                expirationDate.timeInMillis = expirationDateInMillis
-                val formattedExpirationDate = dateFormat.format(expirationDate.time)
-                formBodyBuilder.add(PARAM_EXPIRATION_DATE, formattedExpirationDate)
-            }
+    private fun createFormBodyBuilder(): FormBody {
```

Comment:
Now, we return the FormBody directly, so its name should not contain `Builder` anymore ->`createFormBody()` What do you think?